.github/workflows/release.yml

```diff
@@ -1,26 +1,116 @@
+        with:
+          fetch-depth: 0
+      - name: Create & publish tag
+        id: release
         run: |
-          echo ::set-output name=tag::${GITHUB_REF/refs\/tags\/}
-      - name: Create & Publish Release
-        uses: release-drafter/release-drafter@v5.12.1
+          currentTag=$(git describe --abbrev=0 --tags)
+          echo "current release tag ${currentTag}"
+
+          nextTag=''
+          releaseType=${{ inputs.releaseType }}
+          if [ $releaseType == "minor" ]; then
+            # increment minor version and reset patch version
+            nextTag=$(echo "${currentTag}" | awk -F. '{OFS="."; $2+=1; $3=0; print $0}')
+          else
+            # increment patch version
+            nextTag=$(echo "${currentTag}" | awk -F. '{OFS="."; $3+=1; print $0}')
+          fi
+
+          if [ ${{ inputs.debug }} == 'true' ]; then
+            echo "running workflow in debug mode, next ${releaseType} tag: ${nextTag}"
+          else
+            git tag $nextTag
+            git push origin $nextTag
+            echo "tag=${nextTag}" >> $GITHUB_OUTPUT
+          fi
+    outputs:
+      releaseTag: ${{ steps.release.outputs.tag }}
+
+  publish-release:
+    name: Publish release
+    needs: publish-tag
+    if: contains(inputs.debug, 'false')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Create & publish release
+        uses: release-drafter/release-drafter@v5.22.0
         with:
-          name: ${{ steps.get_version.outputs.tag }}
-          tag: ${{ steps.get_version.outputs.tag }}
-          version: ${{ steps.get_version.outputs.tag }}
+          name: ${{ needs.publish-tag.outputs.releaseTag }}
+          tag: ${{ needs.publish-tag.outputs.releaseTag }}
+          version: ${{ needs.publish-tag.outputs.releaseTag }}
           publish: true
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  publish-docker-image:
+    name: Publish docker image
+    needs: publish-tag
+    if: contains(inputs.debug, 'false')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build image
+        run: |
+          docker build --build-arg DEPLOY_VERSION=${{ needs.publish-tag.outputs.releaseTag }} -t docker.io/prebid/prebid-cache:${{ needs.publish-tag.outputs.releaseTag }} .
+      - name: Login to docker Hub
+        if: contains(inputs.debug, 'false')
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Publish to docker Hub
+        run: |
+          docker push docker.io/prebid/prebid-cache:${{ needs.publish-tag.outputs.releaseTag }}
```

Comment:
Consider swapping the `publish-release` and `publish-docker-image` jobs. Should we wait to publish the release notes until the image has been built + uploaded?

Reply:
Swapped publish-release and publish-docker-image jobs.  Also updated yml to not publish the release notes until the image has been built and uploaded.

Changes are made in https://github.com/prebid/prebid-cache/pull/135/commits/907e5d1c8302fc750406a79546f86cb460d07abc